lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -124,3 +129,86 @@ def get(self, request, program_key=None):
         paginated_enrollments = self.paginate_queryset(enrollments)
         serializer = ProgramEnrollmentListSerializer(paginated_enrollments, many=True)
         return self.get_paginated_response(serializer.data)
+
+    ERROR_CONFLICT = 'conflict'
+    ERROR_DUPLICATED = 'duplicated'
+    ERROR_INVALID_STATUS = 'invalid-status'
+
+    @verify_program_exists
+    def post(self, request, *args, **kwargs):
+        """
+        This is the POST for ProgramEnrollments
+        """
+        if len(request.data) > 25:
+            return Response(
+                status=status.HTTP_413_REQUEST_ENTITY_TOO_LARGE,
+                content_type='application/json',
+            )
+
+        program_uuid = kwargs['program_key']
+        student_data = OrderedDict((
+            row.get('external_user_key'),
+            {
+                'program_uuid': program_uuid,
+                'curriculum_uuid': row.get('curriculum_uuid'),
+                'status': row.get('status'),
+                'external_user_key': row.get('external_user_key'),
+            })
+            for row in request.data
+        )
+
+        total_student_keys = [enrollment.get('external_user_key') for enrollment in request.data]
+        key_counter = Counter(total_student_keys)
+
+        response_data = {}
+        for student_key, count in key_counter.items():
+            if count > 1:
+                response_data[student_key] = self.ERROR_DUPLICATED
+                student_data.pop(student_key)
+
+        existing_enrollments = ProgramEnrollment.bulk_read_by_student_key(program_uuid, student_data)
```

Comment:
Does it actually? It would if this function could be pure, but as is it would be taking a function that is responsible for a bunch of side effects and move it to a place where I can't read it. Perhaps we can talk about how you recommend doing this.

Reply:
If you don't find moving into its own function helps the code, then don't do it. My suggestion is speculative. Add more comments is fine here.